src/styles/theme/button.ts

```diff
@@ -16,75 +16,209 @@ const baseStyles = {
   '&[data-is-loading]': { cursor: 'wait' },
 };
 
+const primary = {
+  px: 3,
+  py: '6/4',
+  color: 'primary50',
+  bg: 'primary600',
+  borderColor: 'primary600',
+  boxShadow: 'sm',
+
+  lineHeight: '130%',
+  '&:hover': {
+    color: 'primary50',
+    bg: 'primary700',
+    borderColor: 'primary700',
+  },
+
+  ...baseStyles,
+  '&:active': {
+    color: 'primary50',
+    bg: 'primary600',
+    borderColor: 'primary200',
+    borderWidth: '4px',
+    borderStyle: 'solid',
+    boxShadow: 'none',
+  },
+  '&:disabled': { cursor: 'not-allowed', bg: 'primary600', opacity: 0.25 },
+  fontWeight: 'bold',
+  fontSize: '12px',
+};
+
+const outline = {
+  px: 5,
+  py: '6/4',
+  color: 'grey900',
+  bg: 'transparent',
+  border: '1px',
+  borderColor: 'grey900',
+  boxShadow: 'sm',
+  '&:hover, &:active': {
+    color: 'white',
+    bg: 'grey900',
+  },
+  ...baseStyles,
+};
+
+const secondary = {
+  px: 5,
+  py: '6/4',
+  color: 'primary700',
+  bg: 'primary200',
+  borderColor: 'primary200',
+  boxShadow: 'sm',
+  lineHeight: '130%',
+  '&:hover': {
+    color: 'primary800',
+    bg: 'primary300',
+    borderColor: 'primary300',
+  },
+
+  ...baseStyles,
+  '&:active': {
+    color: 'primary700',
+    bg: 'primary200',
+    borderColor: 'primary100',
+    borderWidth: '4px',
+    borderStyle: 'solid',
+    boxShadow: 'none',
+  },
+  '&:disabled': {
+    cursor: 'not-allowed',
+    color: 'primary700',
+    bg: 'primary200',
+    opacity: 0.25,
+  },
+  fontWeight: 'bold',
+  fontSize: '12px',
+};
+const underline = {
+  px: 0,
+  pt: 0,
+  pb: 1,
+  color: 'grey900',
+  bg: 'transparent',
+  borderColor: 'transparent',
+  '&::before': {
+    content: JSON.stringify(''),
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    right: 0,
+    height: '2px',
+    bg: 'primary500',
+    transition: 'background-color 0.2s, left 0.2s, right 0.2s',
+  },
+  '&:hover, &:active': {
+    color: 'primary500',
+    '&::before': {
+      left: '50%',
+      right: '50%',
+    },
+  },
+  ...baseStyles,
+};
+
 const buttonTheme = {
   buttons: {
     primary: {
-      px: 5,
-      py: '6/4',
-      color: 'grey900',
-      bg: 'primary500',
-      borderColor: 'primary500',
-      boxShadow: 'sm',
-      '&:hover, &:active': {
-        color: 'white',
-        bg: 'grey900',
-        borderColor: 'grey900',
+      ...primary,
+      small: {
+        ...primary,
+        fontSize: 1,
+      },
+      medium: {
+        ...primary,
+        fontSize: 3,
+        lineHeight: 'medium',
+      },
+      large: {
+        ...primary,
+        fontSize: 5,
+        px: 4,
+        lineHeight: 'high',
       },
-      ...baseStyles,
     },
     outline: {
```

Comment:
Hover effect + inactive styling
<img width="886" alt="Screenshot 2021-02-08 at 14 36 54" src="https://user-images.githubusercontent.com/26187796/107227022-2a396d80-6a1b-11eb-8990-3433510bdf59.png">

Reply:
same goes for the link component (link.ts)

Reply:
please reuse the current hover effect i have on this but reverse it, with the smooth transition of the line

Reply:
I'm not 100% understanding what you mean by "Hover effect + inactive styling" "same goes for the link component (link.ts)". can you please clarify?